src/main/kotlin/org/rust/ide/refactoring/inlineFunction/RsInlineFunctionProcessor.kt

```diff
@@ -409,6 +405,23 @@ private class InlineSingleUsage(
         return declarationLastUsages.any { it is RsPathExpr && it.path == usage }
     }
 
+    private fun prepareFunctionCall(usage: CallUsage): RsExpr {
+        val functionCall = when (usage) {
+            is FunctionCallUsage -> usage.functionCall
+            is MethodCallUsage -> usage.methodCall.parent as RsDotExpr
+        }
+        return functionCall.unwrapAwait() ?: functionCall
+    }
+
+    // `foo().await` -> `foo()`
+    //  ~~~~~ [this]
+    private fun RsExpr.unwrapAwait(): RsExpr? {
+        if (!originalFunction.isAsync) return null
```

Comment:
Maybe it worth it to add a test with awaited non-async function (which returns `impl Future<>`) explicitly?